visualization-design.typ

```diff
@@ -111,7 +111,45 @@ The load balancer is one part of a distributed system, and many instances of the
 
 Grafana supports querying Prometheus, and there is documentation linked below on a quick start for creating a new data source. Given the balancer has already integrated Prometheus for metrics, this is the preferred method for gathering data. There is no mention on compatibility with D3.js, but assuming there are no issues integrating D3.js into a Grafana panel, this should not be a problem. #cite(<grafana-prometheus-visualization>)
 
-=== Setup Instructions From Grafana Documentation
+== Recieving Information From Load Balancers
+
+Grafana is a tool primarily meant for time series data, and no current data source plugins support recieving the type of information needed. To add support a custom data source must be created
+
+=== Balancer Discovery
+
+While the visualization is running: Multiple instances of the balancer can be active simultaneously, new instances can become active, and instances can go offline. The addresses of these balancers are not known at runtime, so a discovery process similiar to @Chapter::MonolithDiscovery must run to collect data from the discovered balancers.
+
+In order to achieve this, a new rust crate will be created to handle this discovery process. Implementation will likely be similar to @Chapter::MonolithDiscovery. A port will be opened to listen for active instances of the balancer. When a connection or connections are found, the balancer discoverer clones the balancer(s) and connects.
```

Comment:
> A port will be opened to listen for active instances of the balancer.

This is not how monolith discovery works. Monolith discovery collects a list of network addresses by polling some external service or receiving the list from some external service.

> the balancer discoverer clones the balancer(s) and connects

Can you elaborate a little more on what you mean here? Why are we cloning balancers?

Reply:
would fly.io be the one to get polled for balancer discovery?

Reply:
Not necessarily. Go read the monolith discovery section